File: recursion/Reverse/rever.py
s='hello'
#using iteration
def rev(n):
    li=[]
    for i in range(len(s)-1,-1,-1):
        li.append(n[i])
    return ''.join(li)
print(rev(s))
#using recursion
def rever2(n):
    if n=='':
        return ''
    else:
        return rever2(n[1:])+n[0]
print(rever2(s))
#using closure 
def rever3(n):
    arr=list(n[0:])
    revarr=[]
    def addtoarr(a):
        if(len(a)>0):
            revarr.append(a.pop())
            addtoarr(a)
    addtoarr(arr)
    return ''.join(revarr)
print(rever3(s))

File: recursion/Reverse/rever.js
function rev(s){
    if (s===''){
        return ''
    }
    else{
        return rev(s.substr(1))+s.charAt(0)
    }
}
console.log(rev('Mom'))

function rev2(s){
    let n=s.length
    let arr=[]
    for(let i=n-1;i>=0;i--){
        arr.push(s[i])

    }
    return arr.join('')
}
console.log(rev2('Mom'))

function rev3(s){
    let ar=s.split('')
    let revarr=[]
    function addtoarr(arr){
        if(arr.length>0){
            revarr.push(ar.pop())
            addtoarr(ar)
        }

    }
    addtoarr(ar)
    return revarr.join('')
}
console.log(rev3('Mom'))